ports/rp2/rp2_pio.c

```diff
@@ -462,21 +462,32 @@ STATIC mp_obj_t rp2_state_machine_init_helper(const rp2_state_machine_obj_t *sel
     }
 
     // Compute the clock divider.
-    float div;
+    uint16_t clkdiv_int;
+    uint8_t clkdiv_frac;
     if (args[ARG_freq].u_int < 0) {
-        div = 1;
+        // Default: run at CPU frequency.
+        clkdiv_int = 1;
+        clkdiv_frac = 0;
     } else if (args[ARG_freq].u_int == 0) {
-        div = 0;
+        // Special case of 0: set clkdiv to 0.
+        clkdiv_int = 0;
+        clkdiv_frac = 0;
     } else {
-        div = (float)clock_get_hz(clk_sys) / (float)args[ARG_freq].u_int;
+        // Frequency given in Hz, compute clkdiv from it.
+        uint64_t div = (uint64_t)clock_get_hz(clk_sys) * 256ULL / (uint64_t)args[ARG_freq].u_int;
+        if (!(div >= 1 * 256 && div <= 65535 * 256)) {
```

Comment:
It's admittedly a bit of a boundary-condition, but according to the RP2040 datasheet I believe that if `div` happens to be _exactly_ `65536 * 256`, then that's achievable by setting both `clkdiv_int` and `clkdiv_frac` to `0`. Also, this check appears to inhibit a `div` value of `65535.5 * 256`, which I believe is perfectly allowable with `clkdiv_int = 65535` and `clkdiv_frac = 128` ?

Reply:
Agreed.  I pushed a change to make this check `div <= 65536 * 256` which should fix both those issues.